src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs

```diff
@@ -275,15 +274,28 @@ private void Flush(bool flushStream, bool flushEncoder)
                 }
             }
 
-            int count = _encoder.GetBytes(_charBuffer, 0, _charPos, _byteBuffer, 0, flushEncoder);
+            // For sufficiently small char data being flushed, try to encode to the stack.
+            // For anything else, fall back to allocating the byte[] buffer.
+            Span<byte> byteBuffer = stackalloc byte[0];
```

Comment:
looks a bit unnatural IMO, perhaps `= default;` is better?

Reply:
It's necessary.  It's the only way to force the necessary lifetime scope on the local to be able to later assign a stackalloc.

Reply:
Oh, good to know, thanks!